File: sprinbwebfluxtutorial/src/main/java/com/qiusuo/webfluxtutorial/reactive/CustomSubscriber.java
package com.qiusuo.webfluxtutorial.reactive;

import org.reactivestreams.Subscription;
import reactor.core.CoreSubscriber;

public class CustomSubscriber implements CoreSubscriber {
    @Override
    public void onSubscribe(Subscription s) {
        s.request(10);
    }

    @Override
    public void onNext(Object o) {
        System.out.println("on Next");
    }

    @Override
    public void onError(Throwable throwable) {
        System.out.println("on error");
    }

    @Override
    public void onComplete() {
        System.out.println("finishing subscribing");
    }
}

File: springwebfluxtutorial/src/main/kotlin/com/qiusuo/webfluxtutorial/reactive/CustomSubscriber.kt
package com.qiusuo.webfluxtutorial.reactive

import org.reactivestreams.Subscription
import reactor.core.CoreSubscriber

class CustomSubscriber : CoreSubscriber<Any> {
    override fun onSubscribe(s: Subscription) {
        s.request(10)
    }

    override fun onNext(o: Any) {
        println("on Next")
    }

    override fun onError(throwable: Throwable) {
        println("on error")
    }

    override fun onComplete() {
        println("finishing subscribing")
    }
}
